web/src/engine/websites/PiccomaFR.ts

```diff
@@ -0,0 +1,116 @@
+import { type ImageLinks, type TileGroup, script } from './Piccoma';
+import { Tags } from '../Tags';
+import icon from './Piccoma.webp';
+import { DecoratableMangaScraper, type MangaPlugin, Manga, Chapter, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchJSON, FetchRequest, FetchWindowScript } from '../FetchProvider';
+import DeScramble from '../transformers/ImageDescrambler';
+import type { Priority } from '../taskpool/TaskPool';
+
+type APIMangas = {
+    data: {
+        p_products: [{
+            product_id: number,
+            title: string,
+        }]
+    }
+}
+
+type NextData = {
+    props: {
+        pageProps: {
+            initialState: {
+                episode: {
+                    episodeList: {
+                        episode_list?: NextChapters
+                    }
+                },
+            }
+        }
+    }
+}
+
+type NextChapters = {
+    id: string,
+    title: string,
+    volume: number,
+    is_free: 'Y' | 'N',
+}[];
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('piccoma-fr', 'Piccoma (French)', 'https://piccoma.com/fr', Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Media.Manga, Tags.Language.French, Tags.Source.Official);
```

Comment:
Not sure if the URL and the regex in `ValidateMangaURL` is correct, because sometimes the website redirects to the subdomain `fr.` (but this was while being region blocked).

Reply:
* https://fr.piccoma.com/fr/product/1188
* https://piccoma.com/fr/product/1188

both works and does not redirect. Better adjusting the regexp then i think.